example/index.html

```diff
@@ -1,23 +1,26 @@
 ---
-layout: default
+layout: page
 ---
 
 <div class="home">
 
   <h1 class="page-heading">Posts</h1>
 
+  {% assign custom_url = site.url | append: site.baseurl %}
+  {% assign full_base_url = site.github.url | default: custom_url %}
```

Comment:
> If the user sets a base url, wouldn't this result in http://username.github.io/project/project/?

What do you mean by `this`? `custom_url` is `site.url + site.baseurl`, and `full_base_url` is `site.github.url` OR `custom_url`, if `site.github.url` doesn't exist.

> Is there any way to do this without assigns?

We need to allow plugins or provide a filter like this in Jekyll or the jekyll-github-metadata plugin. Due to the scope at render time (starts with just the page content, then includes each successive layout in turn), we have to calculate this each time. Alternatively, Jekyll could introduce `site.full_url`.